src/libraries/System.Security.Cryptography.Cose/tests/CoseTestSign1.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+    public sealed class CoseTestSign1
+    {
+        public bool IsEmbedded { get; }
+        private string _label;
+        private Func<CoseTestKey, byte[], byte[]> _signImpl;
+        private Func<CoseTestKey, byte[], CoseSign1Message, bool> _verifyImpl;
+
+        private CoseTestSign1(bool isEmbedded, string label, Func<CoseTestKey, byte[], byte[]> signImpl, Func<CoseTestKey, byte[], CoseSign1Message, bool> verifyImpl)
+        {
+            IsEmbedded = isEmbedded;
+            _label = label;
+            _signImpl = signImpl;
+            _verifyImpl = verifyImpl;
+        }
+
+        public byte[] Sign(CoseTestKeyManager keyManager, string key, byte[] payload)
+        {
+            CoseTestKey signer = keyManager.GetKey(key);
+            return _signImpl(signer, payload);
+        }
+
+        public bool Verify(CoseTestKeyManager keyManager, string expectedKey, byte[] payload, byte[] signature)
+        {
+            CoseSign1Message message = CoseSign1Message.DecodeSign1(signature);
+
+            Assert.True(message.ProtectedHeaders.TryGetValue(CoseHeaderLabel.KeyIdentifier, out CoseHeaderValue keyIdValue));
+            byte[] keyIdBytes = keyIdValue.GetValueAsBytes();
+            string keyId = Encoding.UTF8.GetString(keyIdBytes);
+            if (keyId != expectedKey)
+            {
+                // key identifier doesn't match
+                return false;
+            }
+
+            CoseTestKey key = keyManager.GetKey(expectedKey);
+            return _verifyImpl(key, payload, message);
+        }
+
+        public static IEnumerable<CoseTestSign1> GetImplementations()
+        {
+            yield return new(true, "Sign/VerifyEmbedded(byte[])",
```

Comment:
note for reviewers: this and equivalent in CoseTestMultiSign is a core of testing all overloads